core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -271,24 +277,23 @@ public synchronized void unRegisterDataMapCatalog(DataMapSchema dataMapSchema) {
    */
   public synchronized DataMapCatalog getDataMapCatalog(
       DataMapProvider dataMapProvider,
-      String providerName) throws IOException {
+      String providerName,
+      boolean clearCatalogs) throws IOException {
```

Comment:
This is not clear, maybe it is better to add a separate API to clear the catalog

Reply:
initialy we had a separate API for clear, i have removed in this PR because it was synchronized and instead of calling that API everytime if session changes, it better if we handle here where we intialize catalogs and its synchronized also, so it takes care of all.

Reply:
@jackylk : This is ok, clear is just assigning to null and multiple function calls in concurrent scenario always better to avoid.
However, initializeDataMapCatalogs can be optimized in the future with more testing. we have some duplicate code there to handle the concurrent scenario